substrate/frame/support/procedural/src/pallet/parse/config.rs

```diff
@@ -322,12 +347,30 @@ pub fn replace_self_by_t(input: proc_macro2::TokenStream) -> proc_macro2::TokenS
 		.collect()
 }
 
+/// Check that the trait item requires the `TypeInfo` bound (or similar).
+fn contains_type_info_bound(ty: &TraitItemType) -> bool {
+	const KNOWN_TYPE_INFO_BOUNDS: &[&str] = &[
+		// Explicit TypeInfo trait.
+		"TypeInfo",
+		// Implicit known substrate traits that implement type info.
+		// Note: Aim to keep this list as small as possible.
+		"Parameter",
+	];
+
+	ty.bounds.iter().any(|bound| {
+		let syn::TypeParamBound::Trait(bound) = bound else { return false };
+
+		KNOWN_TYPE_INFO_BOUNDS.iter().any(|known| bound.path.is_ident(known))
```

Comment:
I think that this will not match `scale::TypeInfo` but only `TypeInfo`, isn't it?

What I mean is that we do the match on the whole path, not the last segment, no?

If we want to support only this. Then we can say in the error message of `include_metadata` something like: "Note: only ident is supported, e.g. `scale::TypeInfo` is not detected, only `TypeInfo`.